sycl/source/detail/graph_impl.cpp

```diff
@@ -676,12 +683,17 @@ bool modifiable_command_graph::begin_recording(
   return QueueStateChanged;
 }
 
-bool modifiable_command_graph::end_recording() { return impl->clearQueues(); }
+bool modifiable_command_graph::end_recording() {
+  graph_impl::WriteLock Lock(impl->MMutex);
+  bool ret = impl->clearQueues();
+  return ret;
```

Comment:
```suggestion
  return impl->clearQueues();
```
Doesn't seem like this change was necessary.